android/src/org/ppsspp/ppsspp/PpssppActivity.java

```diff
@@ -112,4 +113,16 @@ public void run() {
 			}
 		});
 	}
+
+	public int openContentUri(String uriString) {
+		try {
+			Uri uri = Uri.parse(uriString);
+			ParcelFileDescriptor filePfd = getContentResolver().openFileDescriptor(uri, "r");
+			int fd = filePfd.detachFd();  // Take ownership of the fd.
+			return fd;
+		} catch (Exception e) {
+			Log.e(TAG, "Exception opening content uri: " + e.toString());
+			return -1;
```

Comment:
Might be nice to differentiate with -2 or something, compared to nativeActivity missing.  Thinking of logging / future exceptions.

-[Unknown]

Reply:
The native activity missing thing is no longer an issue (I think), so I'm not sure it's worth bothering...